server/app/models.py

```diff
@@ -1199,3 +1214,27 @@ def _can(cls, user, need, final, query):
     def _pre_put_hook(self):
         # TODO Remove when submitter is a computed property
         self.submitter = self.submission.get().submitter
+
+    def get_scores(self):
+        """ 
+        Return a list of lists of the format [[student, score, message, grader, tag]]
+        if the submission has been scored. Otherwise an empty list. 
+        If the submission is a group submission, there will be an element 
+        for each combination of student and score.
+        """
+        # TODO: get the most recent score for each tag. 
+        # Question: will all scores have a grader? In particular the scores from the autograder. 
```

Comment:
Ok, then I will have to change this a bit. The current version breaks when there is no grader, but it should be an easy fix.